peekingduck/pipeline/nodes/model/movenetv1/movenet_files/predictor.py

```diff
@@ -161,8 +161,8 @@ def _get_results_multi(
         bbox_masks = self._get_masks_from_bbox_scores(
             bbox_score, self.bbox_score_threshold
         )
-        if len(bbox_masks) == 0:
-            return (np.zeros(0), np.zeros(0), np.zeros(0), np.zeros(0))
+        if not bbox_masks:
+            return np.empty((0, 4)), np.empty(0), np.empty(0), np.empty(0)
```

Comment:
Would changing the structure of the first return object cause problems and/or create confusion?
The original `np.zeros(0)` is an empty array, while the new `np.empty((0,4))` is an empty array of 4 empty slots. Particularly, `np.array_equal(np.zeros(0), np.empty((0,4)))` is `False`.
Analogy in set theory would be {} != {{}} (lhs = empty set, rhs = empty set containing empty set)

Likewise for above `jdev1/jde_files/tracker.py` `online_tlwhs` portion.

Reply:
For typical usages, such as checking if the `bboxes` is empty before performing any postprocessing/visualization tasks, both has the same behavior as shown by the sample code below:
```python
arr_1 = np.empty((0, 4))
arr_2 = np.zeros(0)
print(len(arr_1) == 0, len(arr_2) == 0)
print(arr_1.size == 0, arr_2.size == 0)
print(arr_1.shape[0] == 0, arr_2.shape[0] == 0)

>>> True True
>>> True True
>>> True True
```
Note: using `not arr` to check if a numpy array is empty is not recommended.

However, using `np.empty((0, 4))` has an benefit of allowing some postprocessing steps such as bbox coordinate transformation to proceed without having to do an extra emptiness check. Similar to how you can do 
```python
lst = []
for i in lst:
    print(i)
```
without having to do a `if lst` beforehand. Consider the following code:
```python
def xyxy2tlwh(arr):
    ret = np.empty_like(arr)
    ret[:, 0] = arr[:, 0]
    ret[:, 1] = arr[:, 1]
    ret[:, 2] = arr[:, 2] - arr[:, 0]
    ret[:, 3] = arr[:, 3] - arr[:, 1]

    return ret

arr_1 = np.empty((0, 4))
arr_2 = np.zeros(0)

print(xyxy2tlwh(arr_1))  # prints []
print(xyxy2tlwh(arr_2))  # throws IndexError
```
Ensuring that `bbox` has a guaranteed shape (Nx4, where N>=0) allows the code to run without error